### PR TITLE
Adds Espresso test for shared_preferences

### PR DESCRIPTION
**Pull request by collinjackson**
Demonstrates an Espresso test running on Firebase Test Lab using the new espresso plugin.

### CODE REVIEW

File: packages/shared_preferences/shared_preferences/example/android/app/src/androidTest/java/io/flutter/plugins/sharedpreferencesexample/SharedPreferencesIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            new ActivityTestRule<>(MainActivity.class, true, false);
+
+    @Before
+    public void setUp() {
+        ActivityScenario.launch(MainActivity.class);
+    }
+
+    @Test
+    public void tapToCheckPersistentData() throws Exception {
+        onFlutterWidget(FlutterMatchers.withTooltip("Clear")).perform(FlutterActions.click());
```
Comment (adazh) [inline packages/shared_preferences/shared_preferences/example/android/app/src/androidTest/java/io/flutter/plugins/sharedpreferencesexample/SharedPreferencesIntegrationTest.java]: FYI, you could import these static methods in FlutterMatchers, FlutterActions, FlutterAssertions, so each statement can be shorter.

Comment (collinjackson) [reply packages/shared_preferences/shared_preferences/example/android/app/src/androidTest/java/io/flutter/plugins/sharedpreferencesexample/SharedPreferencesIntegrationTest.java]: sounds good, I'll update the test to do that.

Comment (collinjackson) [reply packages/shared_preferences/shared_preferences/example/android/app/src/androidTest/java/io/flutter/plugins/sharedpreferencesexample/SharedPreferencesIntegrationTest.java]: done

**Comment by stuartmorgan** [comment]
Is there a reason this never landed? More test coverage seems like a good idea 🙂

**Comment by stuartmorgan** [comment]
Closing as obsolete.

**Pull request closed without merging**